enclave/core/arena.c

```diff
@@ -39,11 +39,14 @@ void* oe_arena_malloc(size_t size)
     // Create the anera if it hasn't been created.
     if (_arena.buffer == NULL)
     {
-        void* buffer = oe_allocate_arena(_capacity);
+        _arena.capacity = __atomic_load_n(&_capacity, __ATOMIC_SEQ_CST);
```

Comment:
What is stopping simultaneous reads/writes to _arena?

Reply:
See my comment above.

Reply:
I suppose that's the problem, it can be modified by `oe_configure_arena_capacity` or `oe_arena_malloc` from multiple threads. But then that raises the question, does the assignment to _capacity in `oe_configure_arena_capacity` need to use `__atomic_store_n`?

Reply:
According to the implementation, `oe_arena_malloc` only reads `_capacity`. So I assume the race occurs between `oe_arena_malloc` (read) and `oe_configure_arena_capacity` (write). This also requires `arena.buffer` is not initialized yet.

My follow-up on this is that the design of `oe_configure_arena_capacity` for one-time-use only? Once `arena.buffer` is set, `oe_configure_arena_capacity` seems to become ineffective.